src/pkg/projects/Microsoft.NETCore.App/src/Microsoft.NETCore.App.depproj

```diff
@@ -134,7 +134,32 @@
     </ItemGroup>
     <Error Text="Unexpected System package(s) @(_secondarySystemPackages)" Condition="'@(_secondarySystemPackages)' != ''" />
 
-    <Message Importance="low" Text="%(FilesToPackage.PackagePath) -> @(FilesToPackage)" />
+    <ItemGroup>
+      <!-- pick up any src our sources directory from packages contributing files -->
+      <_sourcePathCandidate Include="@(FilesToPackage->'$(PackagesDir)\%(NuGetPackageId)\%(NuGetPackageVersion)\sources')" />
+      <_sourcePathCandidate Include="@(FilesToPackage->'$(PackagesDir)\%(NuGetPackageId)\%(NuGetPackageVersion)\src')" />
+      <_sourcePath Include="@(_sourcePathCandidate)" Condition="Exists('%(Identity)')" />
+    </ItemGroup>
+
+    <!-- on windows workaround max-path -->
+    <!-- this can be removed once we build on core MSBuild which supports long paths. -->
+    <ItemGroup Condition="'$(OsEnvironment)'=='Windows_NT'">
```

Comment:
I considered it but it felt wrong since mklink is windows only.  I suppose I could make it `'$(MSBuildRuntimeType)' != 'core' and '$(OsEnvironment)'=='Windows_NT'` if you think that is cleaner.